Source/GTMSessionFetcher.m

```diff
@@ -3749,21 +3753,30 @@ - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
  * run after the completion handler. Serial dispatch is guaranteed by dispatching to an intermediate
  * serial queue that targets the actual callback queue (which may or may not be serial).
  */
-- (nonnull dispatch_queue_t)serialCallbackQueue {
+- (nonnull dispatch_queue_t)internalCallbackQueue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
-    return [self serialCallbackQueueUnsynchronized];
+    return [self internalCallbackQueueUnsynchronized];
   }  // @synchronized(self)
 }
 
-- (nonnull dispatch_queue_t)serialCallbackQueueUnsynchronized {
+- (nonnull dispatch_queue_t)internalCallbackQueueUnsynchronized {
   GTMSessionCheckSynchronized(self);
 
-  if (!_serialCallbackQueue) {
-    _serialCallbackQueue = SerialCallbackQueueForTargetQueue(_callbackQueue);
+  if (!_internalCallbackQueue) {
+    // If the fetcher was provided blocks for passive sequential callbacks (not ones where the
+    // system waits for a response, such as the response callback), create a serial queue wrapper
+    // targeting the client-provided callback queue.
+    bool hasSerialCallbacks = self.accumulateDataBlock || self.sendProgressBlock ||
```

Comment:
The only other edge case I can think of is retries.  Is there any way retries could get into trouble with callbacks showing up out of order?

Reply:
I'd been contemplating that, but the retry block is called after the NSURLSessionTask completes with an error; it'll only be called once at at time, and any further action on the fetcher waits until the response block is called, which should provide its own effective serialization.

I guess there is a question about the `resumeData` block, which could be dispatched immediately before the completion block. I'll add that to this check.